folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java

```diff
@@ -69,27 +73,26 @@
 
 public class DefaultRawMemcacheClient extends AbstractRawMemcacheClient {
 
-  private static final DefaultThreadFactory DAEMON_THREAD_FACTORY =
-      new DefaultThreadFactory(DefaultRawMemcacheClient.class, true);
-  private static final EventLoopGroup EVENT_LOOP_GROUP =
-      new NioEventLoopGroup(0, DAEMON_THREAD_FACTORY);
-
   private static final AtomicInteger GLOBAL_CONNECTION_COUNT = new AtomicInteger();
 
   private final Logger log = LoggerFactory.getLogger(DefaultRawMemcacheClient.class);
 
   private final AtomicInteger pendingCounter = new AtomicInteger();
   private final int outstandingRequestLimit;
 
-
   private final Channel channel;
   private final BatchFlusher flusher;
   private final HostAndPort address;
   private final Executor executor;
   private final long timeoutMillis;
   private final int maxSetLength;
 
-  private final AtomicReference<String> disconnectReason = new AtomicReference<>(null);
+  private static final
+    AtomicReferenceFieldUpdater<DefaultRawMemcacheClient, String> DISCONNECT_REASON_UPDATER =
+          AtomicReferenceFieldUpdater.newUpdater(DefaultRawMemcacheClient.class,
+                  String.class, "disconnectReason");
+
+  private volatile String disconnectReason;
```

Comment:
Is this change really worth the added code complexity? The main purpose of `AtomicReferenceFieldUpdater` seems to be to reduce memory usage, but the number of active `DefaultRawMemcacheClient` is typically fairly small.

Reply:
Hi! Thanks for your review! I've just added it to follow Netty style of using references, to reduce memory footprint as much as possible in every case. It's true that client instance count will be usually low, so it's ok for me if you prefer to rollback the change.

Reply:
Yes, I think we would prefer that. If you revert that, we will be happy to squash-merge and make a new release